packages/frontend/components/scaffold-eth/Contract/ReadOnlyFunctionForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { FunctionFragment } from "ethers/lib/utils";
+import { FC, useState } from "react";
+import { useContractRead } from "wagmi";
+import ErrorToast from "~~/components/ErrorToast";
+import { tryToDisplay } from "./displayUtils";
+import InputUI from "./InputUI";
+import { getFunctionInputKey } from "./utils";
+
+const getInitialFormState = (functionFragment: FunctionFragment) => {
+  const initialForm: Record<string, any> = {};
+  functionFragment.inputs.forEach((input, inputIndex) => {
+    const key = getFunctionInputKey(functionFragment, input, inputIndex);
+    initialForm[key] = "";
+  });
+  return initialForm;
+};
+
+interface IFunctionForm {
+  functionFragment: FunctionFragment;
+  contractAddress: string;
+}
+
+export const ReadOnlyFunctionForm: FC<IFunctionForm> = ({ functionFragment, contractAddress }) => {
```

Comment:
Did you see my comments on a previous review about (not) using `FC`?

Reply:
Ohh really sorry, I missed it, I thought you were talking about destructing of props my bad